sunxspex/emission.py

```diff
@@ -18,41 +20,66 @@
 
 np.seterr(all='raise')
 
+def broken_powerlaw(x, p, q, eelow, eebrk, eehigh):
+    """
+    This routine calculates the value of the electron distribution function at electron energy x.
+    The distribution function is normalized so that the integral from eelow to eehigh is 1.
 
-# def bremsstrahlung(elecctron_dist, e_low, e_high, ):
-#     """
-#     Summary
-#
-#     Parameters
-#     ----------
-#     elecctron_dist : `astropy.units.Quantity`
-#         Description
-#     new : TYPE
-#         Description
-#
-#     Returns
-#     -------
-#     photon_flux: `astropy.units.Quantity`
-#         Photon flux in photons s^-1 keV^-1 cm^-2
-#
-#     Notes
-#     _____
-#
-#     This function solves:
-#
-#
-#     """
-#     # mc2 = 510.98d+00
-#     # clight = 2.9979d+10
-#     # au = 1.496d+13
-#     # r0 = 2.8179d-13
-#
-#     photon_flux = 0
-#
-#     return photon_flux
+    Parameters
+    ----------
+    x : np.array
+        electron energy in keV
+    p : float
+        Slope below the break (x < eebrk)
+    q : float
+        Slope above the break (x > eebrk)
+    eelow : float
+        Low cutoff
+    eebrk : float
+        Break
+    eehigh : float
+        High cutoff
+    Returns
+    -------
+    np.array
+        Power law of x
 
+    Notes
+    -----
+    A normalized, double power law electron distribution function is computed.
+    p is the lower power-law index, between eelow and eebrk, and q is the
+    upper power-law index, between eebrk and eehigh.
+
+    History
+    -----
+    03/15/2020, Bin Chen (bin.chen@njit.edu), adapted from SSW Brm2_Distrn.pro:
+        https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_distrn.pro
 
-def broken_powerlaw(x, p, q, eelow, eebrk, eehigh):
+    """
+    # Obtain normalization coefficient, norm.
+    n0 = (q - 1.0) / (p - 1.0) * eebrk ** (p - 1) * eelow ** (1 - p)
+    n1 = n0 - (q - 1.0) / (p - 1.0)
+    n2 = (1.0 - eebrk ** (q - 1) * eehigh ** (1 - q))
+
+    norm = 1.0 / (n1 + n2)
+
+    res = np.zeros_like(x)
+
+    ind0 = np.where(x < eelow)
+    if ind0[0].size > 0:
+        res[ind0] = 0.
+
+    ind1 = np.where((x < eebrk) & (x >= eelow))
+    if ind1[0].size > 0:
+        res[ind1] = norm * n0 * (p - 1.) * x[ind1] ** (-p) * eelow ** (p - 1.)
+
+    ind2 = np.where((x <= eehigh) & (x >= eebrk))
+    if ind2[0].size > 0:
+        res[ind2] = norm * (q - 1.) * x[ind2] ** (-q) * eebrk ** (q - 1.)
+
+    return res
+
+def broken_powerlaw_f(x, p, q, eelow, eebrk, eehigh):
```

Comment:
`broken_powerlaw_f` and `broken_powerlaw` look identical apart from the name?

Reply:
no they are not. broken_powerlaw_f() is the integral form of broken_powerlaw(). Look at, e.g., the power of x, the former has (1-p) and the latter has -p.